scripts/rtl872x.tcl

```diff
@@ -148,24 +150,26 @@ proc rtl872x_init {} {
     global RTL872x_REG_BOOT_CFG
     global RTL872x_REG_LP_KM4_CTRL
     global rtl872x_flash_loader
+    global rtl872x_flash_loader_data
 
     if {[expr {$rtl872x_ready == 0}]} {
         echo "initializing rtl872x"
         init
         reset halt
-        array2mem rtl872x_flash_loader 32 0x00082000 [array size rtl872x_flash_loader]
+        write_memory 0x00082000 32 rtl872x_flash_loader_data
```

Comment:
@avtolstoy the problem you mentioned about "it stucks on init" it's because the old function `array2mem` have to convert the received array into an object and after that it calls to write_memory.
That's also why I couldn't just change the function call, because they receive different data types
See here 👉  https://review.openocd.org/c/openocd/+/6308/8/src/target/startup.tcl#222

I've create a new "flash_loader" with the expected object type and call directly the function and now it works ok.